src/java.base/share/classes/sun/security/util/DerValue.java

```diff
@@ -315,6 +318,9 @@ public DerValue(byte[] encoding) throws IOException {
         }
         int pos = offset;
         tag = buf[pos++];
+        if ((tag & 0x1f) == 0x1f) {
+            throw new IOException("Tag number cannot exceed 30");
```

Comment:
It may be safe if not support multiple bytes tag in the current implementation of JDK, especially the ASN.1 implementation is private.  However, multiple bytes tag is a legal form of ASN.1 encoding, I think.  It would be nice to have a comment to state that this form is not support yet, and we may consider it in the future if needed.  It may be helpful for future code maintenance.

The exception message, "Tag number cannot exceed 30", may be not accuracy.  I think tag number can exceed 30 per the specification, but JDK does not support it yet because we did not run into such tags in practice.  I may use some words like: "Tag number exceed 30 is not supported".

Reply:
Messages updated. "exceed" is a a verb and I'm not sure whether to choose "exceeding" or "that exceeds" so finally use "over". Thanks!